pimd/pim_upstream.c

```diff
@@ -445,6 +457,9 @@ void pim_upstream_join_suppress(struct pim_upstream *up,
 				t_joinsuppress_msec);
 		}
 
+		if (nbr)
+			pim_jp_agg_remove_group(nbr->upstream_jp_agg, up, nbr);
```

Comment:
How does it get added back?

Reply:
Join timer expiry function will put it back(if neighbor exists) after sending join. If it puts back it wont restart the join timer too.